File: mns-reactor/src/main/java/cz/fb/manaus/reactor/betting/MinimalAmountAdviser.java
package cz.fb.manaus.reactor.betting;

import cz.fb.manaus.core.provider.ExchangeProvider;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class MinimalAmountAdviser implements AmountAdviser {

    @Autowired
    private ExchangeProvider provider;

    @Override
    public double getAmount() {
        return provider.getMinAmount();
    }
}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/betting/MinimalAmountAdviser.kt
package cz.fb.manaus.reactor.betting

import cz.fb.manaus.core.provider.ExchangeProvider
import org.springframework.stereotype.Component

@Component
class MinimalAmountAdviser(private val provider: ExchangeProvider) : AmountAdviser {

    override val amount: Double
        get() = provider.minAmount
}
